features/admin.users.v1/components/user-profile/multi-valued-form-field.tsx

```diff
@@ -0,0 +1,644 @@
+     */
+    key: number;
+    /**
+     * Is user managed by parent organization.
+     */
+    isUserManagedByParentOrg?: boolean
+    /**
+     * Profile information.
+     */
+    profileInfo: Map<string, string>;
+    /**
+     * User profile
+     */
+    user: ProfileInfoInterface;
+    /**
+     * Is the field read only.
+     */
+    isReadOnly: boolean;
+    /**
+     * Multi valued attribute values.
+     */
+    multiValuedAttributeValues: Record<string, string[]>;
+    /**
+     * Set multi valued attribute values.
+     */
+    setMultiValuedAttributeValues: (values: Record<string, string[]> | any) => void;
+    verificationEnabled: boolean;
+    verifiedAttributeValueList: string[]
+    fetchedPrimaryAttributeValue: string;
+    primaryAttributeValue: string;
+    verificationPendingValue: string;
+    primaryAttributeSchema: ProfileSchemaInterface;
+    /**
+     * Handle user update callback.
+     */
+    handleUserUpdate: (userId: string) => void;
+    primaryValues: Record<string, string>;
+    setPrimaryValues: (values: Record<string, string> | any) => void;
+    multiValuedInputFieldValue: Record<string, string>;
+    showAttributes: boolean;
+    maxAllowedLimit: number;
+    setIsFormStale: (value: boolean) => void;
+}
+
+const MultiValuedFormField = (props: MultiValuedFormFieldProps) => {
+    const {
+        "data-componentid": componentId = "user-mgt-user-profile",
+        schema,
+        fieldName,
+        key,
+        profileInfo,
+        user,
+        isUserManagedByParentOrg,
+        multiValuedAttributeValues,
+        setMultiValuedAttributeValues,
+        verifiedAttributeValueList,
+        verificationEnabled,
+        fetchedPrimaryAttributeValue,
+        primaryAttributeValue,
+        verificationPendingValue,
+        primaryAttributeSchema,
+        handleUserUpdate,
+        primaryValues,
+        setPrimaryValues,
+        showAttributes,
+        maxAllowedLimit,
+        setIsFormStale,
+        isReadOnly
+    } = props;
+
+    const { t } = useTranslation();
+    const dispatch: Dispatch = useDispatch();
+    const { values } = useFormState();
+
+    const [ multiValuedInputFieldValue, setMultiValuedInputFieldValue ] = useState<Record<string, string>>({});
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+
+    const resolvedComponentId: string = `${ componentId }-${ schema.name }-input`;
+    const resolvedMutabilityValue: string = schema?.profiles?.console?.mutability ?? schema.mutability;
+    const resolvedMultiValueAttributeRequiredValue: boolean
+        = schema?.profiles?.console?.required ?? schema.required;
+    const sharedProfileValueResolvingMethod: string = schema?.sharedProfileValueResolvingMethod;
+    const resolvedPrimarySchemaRequiredValue: boolean
+        = primaryAttributeSchema?.profiles?.console?.required ?? primaryAttributeSchema?.required;
+    const resolvedRequiredValue: boolean = (resolvedMultiValueAttributeRequiredValue
```

Comment:
Shall we consider moving the logic to a custom hook such as useMultiValuedFormField during refactoring?